### PR TITLE
docs(bpmn-workflows): add docs for multi-instance

### PR DESCRIPTION
**Pull request by saig0**
## Description

* add multi-instance section to BPMN workflows
* add markers on the coverage overview

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: docs/src/bpmn-workflows/multi-instance/multi-instance.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+* [Receive Tasks](bpmn-workflows/receive-tasks.html) 
+* [Embedded Sub-Processes](bpmn-workflows/subprocesses.html#embedded-sub-process) 
+
+A multi-instance activity is executed multiple times - once for each element of a given collection (like a _foreach_ loop in a programming language). 
+
+![multi-instance](/bpmn-workflows/multi-instance/multi-instance-example.png) 
+
+On the execution level, a multi-instance activity has two parts: a **multi-instance body** and an inner activity. The multi-instance body is the container for all instances of the inner activity. 
+
+When the activity is entered, the multi-instance body is activated and one instance for every element of the `inputCollection` is created (sequentially or in parallel). When the all instances are completed then the body is completed and the activity is left.
```
Comment (npepinpe) [inline docs/src/bpmn-workflows/multi-instance/multi-instance.md]:
```suggestion
When the activity is entered, the multi-instance body is activated and one instance for every element of the `inputCollection` is created (sequentially or in parallel). When all instances are completed, the body is completed and the activity is left.
```

File: docs/src/bpmn-workflows/multi-instance/multi-instance.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+ 
+In case of a **parallel** multi-instance activity, all instances are created when the multi-instance body is activated. The instances are executed concurrently and independently from each other.    
+
+![parallel multi-instance](/bpmn-workflows/multi-instance/multi-instance-parallel.png) 
+
+## Defining the Collection to Iterate over
+
+A multi-instance activity **must** define the `inputCollection` variable to iterate over (e.g. `items`). The variable is read from the workflow instance on activating the multi-instance body. It must be an `array` of any type (e.g. `["item-1", "item-2"]`). 
+
+In order to access the current element of the `inputCollection` within the instance, the multi-instance activity can define the `inputElement` variable (e.g. `item`). The element is stored as local variable of the instance under the given name.
```
Comment (npepinpe) [inline docs/src/bpmn-workflows/multi-instance/multi-instance.md]:
```suggestion
In order to access the current element of the `inputCollection` within the instance, the multi-instance activity can define the `inputElement` variable (e.g. `item`). The element is stored as a local variable of the instance under the given name.
```

File: docs/src/bpmn-workflows/multi-instance/multi-instance.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+Interrupting and non-interrupting boundary events can be attached to a multi-instance activity.
+
+When an interrupting boundary event is triggered then the multi-instance body and **all active instances** are terminated. The `outputCollection` variable is not propagated to the parent scope (i.e. no partial output).
+
+When an non-interrupting boundary event is triggered then the instances are not affected. The activities at the outgoing path have no access to the local variables since they are bounded to the multi-instance activity.
+
+## Special Multi-Instance Variables
+
+Every instance has a local variable `loopCounter`. It holds the index in the `inputCollection` of this instance, starting by `1`.
```
Comment (npepinpe) [inline docs/src/bpmn-workflows/multi-instance/multi-instance.md]:
```suggestion
Every instance has a local variable `loopCounter`. It holds the index in the `inputCollection` of this instance, starting with `1`.
```

**Review by npepinpe — CHANGES_REQUESTED** [review]
:rocket: 

I like it; I highlighted a few typos/things that seem grammatically wrong (? I'm not native though so don't blindly trust me :sweat_smile:) but everything was quite clear. Then again I was involved in this, so maybe it might be interesting to get feedback from Deepthi or Miguel who may not know very well multi-instance.

One thing, we add images and the BPMN that was used to generate them; I'm wondering if it might not be interesting to try out embedding `bpmn.js` (is there a read-only mode?) and just render the BPMNs directly. Not to do in this PR but just a thought

**Comment by saig0** [comment]
@npepinpe , I applied your suggestions ;)

I'll look into the BPMN rendering afterwards :+1:

**Review by npepinpe — APPROVED** [review]

**Pull request merged**